Sources/TensorFlow/Layers/Core.swift

```diff
@@ -144,14 +143,14 @@ public struct Dense<Scalar: TensorFlowFloatingPoint>: Layer {
     public var weight: Tensor<Scalar>
     /// The bias vector.
     public var bias: Tensor<Scalar>
-    public typealias Activation = @differentiable (Tensor<Scalar>) -> Tensor<Scalar>
+
     /// The element-wise activation function.
-    @noDerivative public let activation: Activation
+    @noDerivative public let activation: Activation<Scalar>
```

Comment:
In the interest of reducing indirections, I'd suggest removing this type alias instead of defining it elsewhere.